### DISCUSSION TITLE
Span with custom traceId

### DISCUSSION THREAD

**Discussion by xsmrcek** [open-telemetry/opentelemetry-java-instrumentation#5403] [Q&A]
Hello,
Is there any wsy to start custom span with custom traceId and spanId value? I have to do a little custom instrumentation for part of my code which is instrumented by opentelemetry agent or application insights agents (I am testing them both). I did find a way how to start custom span, but I wasn;t able to find out how to set custom trace info. it is possible somehow please?

**Comment by trask** [comment]
hey @xsmrcek! it takes a few steps:

* create a SpanContext from custom traceId and spanId values
* then create a Span using Span.wrap(SpanContext)
* then create a Context using Context.root().with(span)
* then call setParent(Context) on your SpanBuilder

another option if you have a traceparent is to use W3CTraceContextPropagator.getInstance()extract(Context.root(), ...) to create the Context that you feed to SpanBuilder.setParent()

**Comment by xsmrcek** [comment]
hello @trask 
Thank you! With this guide I was able to implement it :)
